packages/core/src/mixins/anchored.js

```diff
@@ -207,26 +207,23 @@ const anchored = ({
                     nextState.position.left =
                         anchorOffset.left + anchorRect.width - selfRect.width;
 
-                    // Calibrate arrow position to stay in safe area
-                    nextState.arrowPosition.left = Math.max(
-                        // anchorOffset.left - nextState.position.left + anchorHalfWidth
-                        selfRect.width - anchorHalfWidth,
-                        edgePadding
-                    );
+                    // anchorOffset.left - nextState.position.left + anchorHalfWidth
+                    nextState.arrowPosition.left = selfRect.width - anchorHalfWidth;
                     break;
 
                 // Left-align to the anchor
                 default:
                     nextState.position.left = anchorOffset.left;
-
-                    // Calibrate arrow position to stay in safe area
-                    nextState.arrowPosition.left = Math.min(
-                        anchorHalfWidth,
-                        arrowSafeAreaWidth
-                    );
+                    nextState.arrowPosition.left = anchorHalfWidth;
                     break;
             }
 
+            // Calibrate arrow position so it stays in safe area
+            if (nextState.arrowPosition.left < edgePadding
+                || nextState.arrowPosition.left > arrowSafeAreaWidth) {
+                delete nextState.arrowPosition;
```

Comment:
If the anchored component place in the right edge, the `arrowPosition` will be removed.

If remove calculated arrow position: 
![screen shot 2017-10-26 at 5 30 45 pm](https://user-images.githubusercontent.com/5558360/32045708-8f692d9e-ba06-11e7-8a8b-d51d2f7667de.png)

Remains arrow position:
![screen shot 2017-10-26 at 5 30 30 pm](https://user-images.githubusercontent.com/5558360/32045726-9d4bc20a-ba06-11e7-845f-7a6360e661f9.png)

Reply:
If revert the `arrowPosition`, the arrow float to right when the popover append on <Text> component XDD

![screen shot 2017-10-26 at 5 36 02 pm](https://user-images.githubusercontent.com/5558360/32045937-33d31d5e-ba07-11e7-8866-47af0169c451.png)